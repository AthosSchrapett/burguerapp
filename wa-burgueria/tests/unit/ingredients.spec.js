import mockAxios from "jest-mock-axios";
import IngredientsService from "@/services/ingredients";

describe("IngredientsService 🍖 🍔 🧀", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should return a list of ingredients from backend", async () => {
    const ingredientes = {
      "paes": [
        {
          "id": 1,
          "tipo": "Italiano Branco"
        },
        {
          "id": 2,
          "tipo": "3 Queijos"
        },
        {
          "id": 3,
          "tipo": "Parmesão e Orégano"
        },
        {
          "id": 4,
          "tipo": "Integral"
        }
      ],
      "carnes": [
        {
          "id": 1,
          "tipo": "Maminha"
        },
        {
          "id": 2,
          "tipo": "Alcatra"
        },
        {
          "id": 3,
          "tipo": "Picanha"
        },
        {
          "id": 4,
          "tipo": "Veggie burger"
        }
      ],
      "opcionais": [
        {
          "id": 1,
          "tipo": "Bacon"
        },
        {
          "id": 2,
          "tipo": "Cheddar"
        },
        {
          "id": 3,
          "tipo": "Salame"
        },
        {
          "id": 4,
          "tipo": "Tomate"
        },
        {
          "id": 5,
          "tipo": "Cebola roxa"
        },
        {
          "id": 6,
          "tipo": "Pepino"
        }
      ]
    }
    mockAxios.get.mockImplementationOnce(() => {
      return Promise.resolve({ data: { ingredientes }})
    });

    const response = await IngredientsService(mockAxios).getIngredients();

    expect(response.data).toHaveProperty('ingredientes');
    expect(response.data).toMatchSnapshot();
  });

  it("should return an error when not found", async () => {
    const errors = { status: 404, statusText: 'Not Found'}
    mockAxios.get.mockImplementationOnce(() => {
      return Promise.resolve({ request: errors})
    })

    const response = await IngredientsService(mockAxios).getIngredients();

    expect(response.errors).toHaveProperty("status");
    expect(response.errors).toHaveProperty("statusText");
  })
});
