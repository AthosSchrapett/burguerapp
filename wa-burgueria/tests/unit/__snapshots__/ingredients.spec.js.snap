// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngredientsService üçñ üçî üßÄ should return a list of ingredients from backend 1`] = `
Object {
  "ingredientes": Object {
    "carnes": Array [
      Object {
        "id": 1,
        "tipo": "Maminha",
      },
      Object {
        "id": 2,
        "tipo": "Alcatra",
      },
      Object {
        "id": 3,
        "tipo": "Picanha",
      },
      Object {
        "id": 4,
        "tipo": "Veggie burger",
      },
    ],
    "opcionais": Array [
      Object {
        "id": 1,
        "tipo": "Bacon",
      },
      Object {
        "id": 2,
        "tipo": "Cheddar",
      },
      Object {
        "id": 3,
        "tipo": "Salame",
      },
      Object {
        "id": 4,
        "tipo": "Tomate",
      },
      Object {
        "id": 5,
        "tipo": "Cebola roxa",
      },
      Object {
        "id": 6,
        "tipo": "Pepino",
      },
    ],
    "paes": Array [
      Object {
        "id": 1,
        "tipo": "Italiano Branco",
      },
      Object {
        "id": 2,
        "tipo": "3 Queijos",
      },
      Object {
        "id": 3,
        "tipo": "Parmes√£o e Or√©gano",
      },
      Object {
        "id": 4,
        "tipo": "Integral",
      },
    ],
  },
}
`;
